### PR TITLE
update installation documentation for tardis

### LINKED ISSUE #1591: Installation Instructions on the Documentation are Outdated
**Describe the bug**

The current instructions for installation of TARDIS package are old are outdated & needs to be updated.
These can be seen [here](https://tardis-sn.github.io/tardis/installation.html#installation-troubles-faq)
Some potential changes may include TARDIS not supporting Python 2.7, changes for the sphinx builds etc. 

**System**
 - OS:
    - [x] GNU/Linux
    - [ ] macOS

### PR DESCRIPTION
**Pull request by erinvisser**
Removed conda forge for non-developers, switched order of developers and non-developers

Fixes #1591 #1924 

### :pencil: Description

**Type:** :memo: `documentation`

### CODE REVIEW

File: docs/installation.rst [hunk 1]
```diff
@@ -78,7 +56,24 @@ this method by following the steps described below.
...
+
+      ::
 
+        $ pip install git+https://github.com/tardis-sn/tardis.git@{tag}
+
+      For example- 
+
+      ::
+      
+        $ pip install git+https://github.com/tardis-sn/tardis.git@release-2023.09.17
```
Comment (andrewfullard) [inline docs/installation.rst:70]: The release tag chosen needs to explicitly match the release tag of the environment file to guarantee compatibility. This should be made very clear in the docs.

File: docs/installation.rst [hunk 2]
```diff
@@ -39,29 +39,7 @@ this method by following the steps described below.
...
-        $ conda install tardis-sn --channel conda-forge --no-deps
-
-      or trying the most recent, unreleased changes from upstream.
-
-      ::
-
-        $ pip install git+https://github.com/tardis-sn/tardis.git@master
-
-   b. Instead, developers should `fork the repository <https://github.com/tardis-sn/tardis/fork>`_, configure
+3. a. Developers should `fork the repository <https://github.com/tardis-sn/tardis/fork>`_, configure
```
Comment (AlexHls) [inline docs/installation.rst:42]: There should be a somewhere that the conda.lock file downloaded in 1. from the latest (or really any) releasemight not match the conda.lock file from the current master branch/ the forked repository. See issue #2429

Developers should imo create their conda environment using the conda lockfile from the cloned/ forked tardis repository, not the release.

**Review by AlexHls — CHANGES_REQUESTED** [review]

File: docs/installation.rst (hunk 1 above)
Comment (Knights-Templars) [reply docs/installation.rst:70]:
```suggestion
        $ pip install git+https://github.com/tardis-sn/tardis.git@release-<yyyy.mm.dd> 
        
        Please fill in the year, month, and date of the released version you want to use.
```

File: docs/installation.rst [hunk 3]
```diff
@@ -78,7 +56,26 @@ this method by following the steps described below.
...
+        
+        Please fill in the year, month, and date of the released version you want to use.
+
+      or trying the most recent, unreleased changes from upstream.
 
+      ::
+
+        $ pip install git+https://github.com/tardis-sn/tardis.git@master
 
 4. Once finished working, you can deactivate your environment.
```
Comment (DeerWhale) [inline docs/installation.rst:81]: Please add "$ wget -q https://github.com/tardis-sn/tardis/releases/latest/download/conda-{platform}-64.lock" in line 110 with the same indentation with the line below that "$ conda update xxx"

File: docs/installation.rst (hunk 2 above)
Comment (DeerWhale) [reply docs/installation.rst:42]: Please change line 33 to "$ wget -q https://github.com/tardis-sn/tardis/blob/master/conda-{platform}-64.lock"

**Review by DeerWhale — APPROVED** [review]

**Review by andrewfullard — APPROVED** [review]

**Pull request merged**